pkg/didcomm/protocol/didexchange/persistence_test.go

```diff
@@ -118,24 +124,156 @@ func TestConnectionRecorder_GetInvitation(t *testing.T) {
+		require.Equal(t, storedConnRec, connRec)
+	})
+	t.Run("test error", func(t *testing.T) {
 		store := &mockstorage.MockStore{Store: make(map[string][]byte), ErrGet: fmt.Errorf("get error")}
 		record := NewConnectionRecorder(store)
 		require.NotNil(t, record)
-		require.NoError(t, store.Put("key1", []byte("value1")))
-		_, err := record.GetConnectionRecord("key1")
+		connRecBytes, err := json.Marshal(&ConnectionRecord{ConnectionID: connIDValue, ThreadID: threadIDValue,
+			Namespace: myNSPrefix})
+		require.NoError(t, err)
+		connKey := connectionKeyPrefix(connIDValue)
+		require.NoError(t, store.Put(connKey, connRecBytes))
+		_, err = record.GetConnectionRecord(connIDValue)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "get error")
+	})
+}
+
+func TestConnectionRecorder_SaveConnectionRecord(t *testing.T) {
+	t.Run("save connection record and get connection Record success", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		connRec := &ConnectionRecord{ThreadID: threadIDValue,
+			ConnectionID: connIDValue, State: stateNameInvited, Namespace: theirNSPrefix}
+		err := record.saveNewConnectionRecord(connRec)
+		require.NoError(t, err)
+
+		storedRecord, err := record.GetConnectionRecord(connRec.ConnectionID)
+		require.NoError(t, err)
+		require.Equal(t, connRec, storedRecord)
+	})
+	t.Run("save connection record and fetch from no namespace error", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		connRec := &ConnectionRecord{ThreadID: threadIDValue,
+			ConnectionID: connIDValue, State: stateNameInvited}
+		err := record.saveNewConnectionRecord(connRec)
 		require.Error(t, err)
+		require.Contains(t, err.Error(), "empty")
+	})
+	t.Run("save connection record error", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte), ErrPut: fmt.Errorf("get error")}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		connRec := &ConnectionRecord{ThreadID: "",
+			ConnectionID: "test", State: stateNameInvited, Namespace: theirNSPrefix}
+		err := record.saveConnectionRecord(connRec)
+		require.Contains(t, err.Error(), "get error")
+	})
+	t.Run("save connection record error", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte), ErrPut: fmt.Errorf("get error")}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+		connRec := &ConnectionRecord{ThreadID: threadIDValue,
+			ConnectionID: connIDValue, State: stateNameInvited, Namespace: theirNSPrefix}
+		err := record.saveNewConnectionRecord(connRec)
 		require.Contains(t, err.Error(), "get error")
 	})
 }
+func TestConnectionRecorder_GetConnectionRecordByNSThreadID(t *testing.T) {
+	t.Run(" get connection record by protocol ID in my name space", func(t *testing.T) {
```

Comment:
seems like some instances of "protocol ID" haven't been replaced.

Reply:
replaced